proposals/mutate_existing_resources.md

```diff
@@ -0,0 +1,213 @@
+    - To *"mutate trigger resources based on policy update"*, the syntax is slightly different in order to simplify policy definition, see example 3 below.
+- A new attribute `spec.rules.mutate.targets` will be introduced to specify a list of target resources to be mutated
+- Kyverno will fetch and mutate the existing resources once the trigger resource changes, the user needs to grant Kyverno permissions to operate on target resources
+- The patches will be added as an annotation to existing resources.
+
+
+
+
+## Examples
+
+1. This exmaple adds the label `foo=bar` to deployment `staging/example-A` on configmap `staging/dictionary` update.
+
+```yaml=
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: "..."
+spec:
+  rules:
+    - name: "..."
+      match:
+        resources:
+          kinds:
+          - ConfigMap
+          names:
+          - dictionary
+          namespaces:
+          - staging
+      mutate:
+        # specify target resources to be mutated
+        # 
+        # if this tag is specificied, Kyverno will 
+        # automatically perform post mutation, there's 
+        # no need to enable this feature additionally
+        targets:
+          # starts with GVK, can be extended to suppport
+          # labels, annotations, etc
+        - kinds:
+          - Deployment
+          names:
+          - example-A
+          namespaces:
+          - staging
+        patchStrategicMerge:
+          metadata:
+            labels:
+              foo: bar
+```
+
+2. This exmaple restarts the deployment `staging/example-A` by updating the timestamp annotation on secret update.
+```yaml=
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: "..."
+spec:
+  rules:
+    - name: "..."
+      match:
+        resources:
+          kinds:
+          - Secret
+          names:
+          - certificate-store
+          namespaces:
+          - staging
+      mutate:
+        targets:
+        - kinds:
+          - Deployment
+          names:
+          - example-A
+          namespaces:
+          - staging
+        patchStrategicMerge:
+          metadata:
+            annotations:
+              kyverno.io/since-last-update: "{{ time_since('', '{{ request.object.metadata.creationTimestamp }}', '') }}"
+```
+
+3. This exmaple adds label `foo=bar` to both incoming and existing deployments on policy updates.
+```yaml=
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: "..."
+spec:
+  rules:
+    - name: "..."
+      match:
+        resources:
+          kinds:
+          - Deployment
+      mutate:
+        # mutateExisting is used to mutate the same 
+        # resources as specified in the match block
+        # the trigger resource is clusterpolicy/policy only
+        mutateExisting: true
+        # targets:
+        patchStrategicMerge:
+          metadata:
+            annotations:
+              foo: bar
+```
+
+4. This exmaple updates the threshold in the configmap upon pod creation.
```

Comment:
@JimBugwadia - can you help further clarify this use case? Or is there any link for this request?